src/scenarios/mauiblazorandroid/pre.py

```diff
@@ -1,49 +1,99 @@
 
-    with ZipFile(apknamezip) as zip:
-        zip.extractall(os.path.join('.', PUBDIR))
+precommands = PreCommands()
+precommands.install_workload('maui', ['--configfile', 'MauiNuGet.config'])
 
-    assets_dir = os.path.join(PUBDIR, 'assets')
-    assets_zip = os.path.join(assets_dir, 'assets.zip')
-    with ZipFile(assets_zip) as zip:
-        zip.extractall(assets_dir)
+# Setup the Maui folder
+precommands.new(template='maui-blazor',
+                output_dir=const.APPDIR,
+                bin_dir=const.BINDIR,
+                exename=EXENAME,
+                working_directory=sys.path[0],
+                no_restore=False)
 
-    os.remove(assets_zip)
-else:
-    copyfile(apkname, os.path.join(PUBDIR, apkname))
+# Add the index.razor.cs file
+with open(f"{const.APPDIR}/Pages/Index.razor.cs", "w") as indexCSFile:
+    indexCSFile.write('''
+    using Microsoft.AspNetCore.Components;
+    #if ANDROID
+        using Android.App;
+    #endif\n\n''' 
+    + f"    namespace {EXENAME}.Pages" + 
+'''
+    {
+        public partial class Index
+        {
+            protected override void OnAfterRender(bool firstRender)
+            {
+                if (firstRender)
+                {
+                    #if ANDROID
+                        var activity = MainActivity.Context as Activity;
+                        activity.ReportFullyDrawn();
+                    #else
+                        System.Console.WriteLine(\"__MAUI_Blazor_WebView_OnAfterRender__\");
+                    #endif
+                }
+            }
+        }
+    }
+''')
+
+# Replace line in the Android MainActivity.cs file
+with open(f"{const.APPDIR}/Platforms/Android/MainActivity.cs", "r") as mainActivityFile:
+    mainActivityFileLines = mainActivityFile.readlines()
+
+with open(f"{const.APPDIR}/Platforms/Android/MainActivity.cs", "w") as mainActivityFile:
+    for line in mainActivityFileLines:
+        if line.startswith("{"):
+            mainActivityFile.write("{\npublic static Android.Content.Context Context { get; private set; }\npublic MainActivity() { Context = this; }")
+        else:
+            mainActivityFile.write(line)
+
+# Build the APK
+precommands.execute(['--no-restore', '--source', 'MauiNuGet.config'])
 
+# Remove the aab files as we don't need them, this saves space
+output_file_partial_path = f"com.companyname.{str.lower(EXENAME)}"
+if args.output_dir:
+    output_file_partial_path = os.path.join(args.output_dir, output_file_partial_path)
 
+os.remove(f"{output_file_partial_path}-Signed.aab")
+os.remove(f"{output_file_partial_path}.aab")
 
+# Copy the MauiVersion to a file so we have it on the machine
+result = subprocess.run(['powershell', '-Command', rf'Get-ChildItem .\{const.APPDIR}\obj\Release\net7.0-android\android-arm64\linked\Microsoft.Maui.dll | Select-Object -ExpandProperty VersionInfo | Select-Object ProductVersion | Select-Object -ExpandProperty ProductVersion'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
```

Comment:
There is a lot of duplication here in mauiandroid, mauiandroidpodcast, and mauiblazorandroid. Is it possible instead to store the powershell command as a separate script, then run that script from python instead? that way we don't need to update the script in multiple places.

Reply:
I agree. And even parts of Python code are duplicated in `pre.py`/`test.py`. This could be refactored, once in final stretch, to separate files.